### PR TITLE
Github action for uploading static assets to release

### LINKED ISSUE #3163: Add a Github action to build static files and add it as a asset to the release page
## Problem
We will be building various packages like a `.deb` file during release which will be used for installing Mathesar. These packages need compiled static assets along with the source code. To be consistent we need to maintain a single source of truth for the source which will be used for various packages.

## Proposed solution
When we make a release, we can make use of GitHub actions to build the static files from the release branch, zip them along with other source code files, and add them to the release assets.
github com_centerofci_mathesar_releases_tag_0 1 2

### PR DESCRIPTION
**Pull request by Anish9901**
Fixes #3163

**Technical details**

- This action gets triggered any time a release gets published or edited.
- This will add a .zip to the release as an asset and will be named e.g. as `mathesar-v0.1.3.zip`

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Anish9901** [comment]
[This](https://github.com/centerofci/mathesar/blob/2726d9b440bf3206b4fa57d7c127ded7ed512d53/config/settings/common_settings.py#L238 ) causes `python manage.py collectstatic` to look into the dir `mathesar/static/mathesar/` but this dir is [ignored](https://github.com/centerofci/mathesar/blob/2726d9b440bf3206b4fa57d7c127ded7ed512d53/.gitignore#L189) in `.gitignore` so it causes an error.
Possible fixes:
- The one done in this PR by adding the folder.
- Adding that dir during release with appropriate assets.

What do you think @silentninja?

File: .github/workflows/create-static-assets-on-release.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          python-version: '3.9'
+
+      - name: Download requirements
+        run: |
+          pip install -r requirements.txt
+          cp .env.example .env
+
+      - name: Run collectstatic
+        run: python manage.py collectstatic --noinput
```
Comment (silentninja) [inline .github/workflows/create-static-assets-on-release.yml:25]: Before running `collectstatic` you need to build the frontend Svelte code in order to include the frontend static files.. Please take a look at the [Docker build steps for building the frontend code](https://github.com/centerofci/mathesar/blob/c714892f3a0529cd65fadca8440b6cc3cb489ec0/Dockerfile#L66)

**Review by silentninja — CHANGES_REQUESTED** [review]
With the current changes, only Django related staticfiles will be collected as the Svelte app is not built. Instead, you would need to install `nodejs` and then build the frontend code before running `collectstatic`

**Comment by mathemancer** [comment]
This is no longer needed, since we're no longer going to do the Debian packaging for this release.

**Pull request closed without merging**